src/Vlingo.Wire.Tests/Fdx/Inbound/InboundStreamTest.cs

```diff
@@ -16,31 +17,35 @@
 
 namespace Vlingo.Wire.Tests.Fdx.Inbound
 {
-    public class InboundStreamTest: IDisposable
+    public class InboundStreamTest : IDisposable
     {
         private TestActor<IInboundStream> _inboundStream;
         private MockInboundStreamInterest _interest;
         private MockChannelReader _reader;
         private TestWorld _world;
 
-        [Fact]
-        public void TestInbound()
+        [Theory]
+        [InlineData(1)]
+        [InlineData(5)]
+        public void TestInbound(int happenings)
         {
-            _interest.TestResult.UntilStops = TestUntil.Happenings(1);
-            while (_reader.ProbeChannelCount.Get() == 0)
-                ;
-            
+            var counter = 0;
+            var accessSafely = AccessSafely.AfterCompleting(happenings)
+                .WritingWith<int>("count", (value) => counter += value)
+                .ReadingWith("count", () => counter);
+            _interest.TestResult.UntilStops = accessSafely;
+
+            ProbeUntilConsumed(() => accessSafely.ReadFrom<int>("count") < 1, _reader);
+
             _inboundStream.Actor.Stop();
-            
+
             int count = 0;
-            foreach (var message in _interest.TestResult.Messages)
-            {
-                ++count;
-                Assert.Equal(MockChannelReader.MessagePrefix + count, message);
-            }
-   
-            _interest.TestResult.UntilStops.Completes();
-    
+            count += (from item in Enumerable.Range(1, _interest.TestResult.Messages.Count)
```

Comment:
We're not using anywhere in the code base this format of linq queries. I would advice either leaving the loop which is more readable in that particular case or use the fluent format `_interest.TestResult.Messages.Where(...prefix).Count()`

I think also that you can join the declaration and assignment like `var count = <query>`

Reply:
Yeah, I will covert it to the loop, query LINQ syntax does not fit in.

Reply:
@tjaskula I've checked the tests and this was the only one using `TestUntil`. A quick search showed that TestUntil is set in the `TestRequestChannelConsumerProvider` class, but it's not referenced anywhere. Should I remove it?

Reply:
> @tjaskula I've checked the tests and this was the only one using `TestUntil`. A quick search showed that TestUntil is set in the `TestRequestChannelConsumerProvider` class, but it's not referenced anywhere. Should I remove it?

Thanks @mihaj. Could you just get rid of the `TestUntil` property in `TestRequestConsumerProvider` ?